src/Headers.php

```diff
@@ -227,13 +218,19 @@ public function addHeaderLine($headerFieldNameOrLine, $fieldValue = null)
         }
 
         if ($fieldValue === null) {
-            $this->addHeader(Header\GenericHeader::fromString($headerFieldNameOrLine));
+            list($fieldName, $fieldValue) = Header\GenericHeader::splitHeaderLine($headerFieldNameOrLine);
+            $norm = $this->normalizeFieldName($fieldName);
+            $this->headersKeys[] = $norm;
+            $this->headers[] = $headerFieldNameOrLine;
```

Comment:
`addHeader()` was previously doing the work you're doing here; why are you duplicating it?